charts/redpanda/console.tpl.go

```diff
@@ -20,10 +20,192 @@ import (
 	"fmt"
 
 	"github.com/redpanda-data/console/backend/pkg/config"
+	"github.com/redpanda-data/helm-charts/charts/console"
 	"github.com/redpanda-data/helm-charts/pkg/gotohelm/helmette"
+	"github.com/redpanda-data/helm-charts/pkg/kube"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/ptr"
 )
 
-func ConsoleConfig(dot *helmette.Dot) any {
+// To connect Console with Redpanda the values are adjusted.
+func renderConsole(dot *helmette.Dot) []kube.Object {
+	v := helmette.UnmarshalInto[Values](dot.Values)
+
+	if !ptr.Deref(v.Console.Enabled, true) {
+		return nil
+	}
+
+	consoleDot := dot.Subcharts["console"]
+
+	consoleValue := helmette.UnmarshalInto[console.Values](consoleDot.Values)
```

Comment:
Let's add a lot of comments to this function. There were a handful previously but we can definitely do better.

Noting that the `console.*` section of the redpanda chart disables the secret/configmap in order for the redpanda chart to render it with connection information will be a great start.